docs/aws/audit/cloudfrontmonitoring/rules/cloudfront_fieldlevel_encryption.mdx

```diff
@@ -23,6 +23,296 @@ SOC2, GDPR, ISO27001, HIPAA, HITRUST, NISTCSF, PCIDSS
+
+   Installation:
+   ```
+   pip install awscli
+   ```
+   Configuration:
+   ```
+   aws configure
+   ```
+   You will be prompted to provide your AWS Access Key ID, Secret Access Key, default region name, and default output format.
+
+2. List CloudFront Distributions: Use the AWS CLI command "list-distributions" to get a list of all the CloudFront distributions. The command is as follows:
+   ```
+   aws cloudfront list-distributions
+   ```
+   This command will return a JSON output with details of all the CloudFront distributions.
+
+3. Check Field-Level Encryption: For each CloudFront distribution, check the "FieldLevelEncryptionId" field. If this field is empty or not present, it means that field-level encryption is not enabled for that distribution. You can do this by parsing the JSON output using a tool like jq. The command is as follows:
+   ```
+   aws cloudfront list-distributions | jq -r '.DistributionList.Items[] | select(.FieldLevelEncryptionId == "")'
+   ```
+   This command will return the details of all the CloudFront distributions where field-level encryption is not enabled.
+
+4. Repeat the process: Repeat steps 2 and 3 for all the AWS accounts and regions in your organization to ensure that field-level encryption is enabled for all CloudFront distributions.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install and configure AWS SDK for Python (Boto3):
+   You need to install Boto3 in your Python environment. You can do this using pip:
+   ```
+   pip install boto3
+   ```
+   After installing Boto3, you need to configure it. You can do this by creating a new session using your AWS credentials:
+   ```python
+   import boto3
+
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
+       aws_session_token='SESSION_TOKEN',
+   )
+   ```
+2. Use Boto3 to list all CloudFront distributions:
+   You can use the `list_distributions` method of the CloudFront client in Boto3 to list all your CloudFront distributions:
+   ```python
+   cloudfront = session.client('cloudfront')
+   distributions = cloudfront.list_distributions()
+   ```
+3. Check each distribution for field-level encryption:
+   You can iterate over each distribution and check if field-level encryption is enabled. If the `FieldLevelEncryptionId` is empty, then field-level encryption is not enabled:
+   ```python
+   for distribution in distributions['DistributionList']['Items']:
+       if not distribution['DefaultCacheBehavior']['FieldLevelEncryptionId']:
+           print(f"Field-level encryption is not enabled for distribution {distribution['Id']}")
+   ```
+4. Run the script:
+   Finally, you can run the script to check all your CloudFront distributions. If any distributions do not have field-level encryption enabled, they will be printed to the console.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>